src/resources/js/v2/tickets-utils.js

```diff
@@ -84,48 +84,67 @@ tribe.tickets.utils = {};
 	 * Removes separator characters and converts decimal character to '.'
 	 * So they play nice with other functions.
 	 *
+	 * @since TBD major refactoring of the internal logic.
 	 * @since 5.0.3
 	 *
+	 * @todo clean up all calls to obj.cleanNumber to do not send a provider.
+	 *
 	 * @param {number} passedNumber The number to clean.
 	 * @param {string} provider The provider.
 	 *
 	 * @returns {string} The cleaned number.
 	 */
-	obj.cleanNumber = function( passedNumber, provider ) {
-		let number = passedNumber;
-		const format = obj.getCurrencyFormatting( provider );
+	obj.cleanNumber = function( passedNumber, provider ) { // eslint-disable-line no-unused-vars
```

Comment:
This should also work with the other providers (Woo for example), and the info about the format is here `const format = obj.getCurrencyFormatting( provider );`. That's not only here but in other places and my fear is that by modifying/removing this we can break how things work for the other providers. Why is it that we're modifying this?

This piece of code is super important to work with the different formatting that people use around the world.

Reply:
@juanfra @bordoni the reason for this change is that the current `cleanNumber'` method fails if it gets passed a value that is formatted differently from what the `provider` says it should. For TC, this means that any values that use dot for thousands and comma for decimals fail. 

You can see this if you apply the patch I suggested here https://gist.github.com/moraleida/4a01dee2a0859421318a649a507e6d39 and try to make a purchase using the Attendee Registration Modal.

Any values from currencies that use dot as the thousands separator get multiplied by 100 in the modal because 
in TC, the values that get passed to this method are already formatted floats. This way, `cleanNumber` tries to remove a thousands separator and ends up removing the decimal separator instead like so:

```
number = '1034.89';
number.split( format.thousands_sep ).join( '' ); // number = '103489'
```

The item price is `R$ 1.034,89`  and the number that shows up on the modal is `R$ 103.489,00`

The logic I applied to this change is the same format-agnostic one we already have in `Price::to_numeric()`, that is: if a non-digit is acting as a decimal, treat it as such, regardless of what the provider says. The only instance in which it would fail is if we're dealing with currencies that use 3-digit decimals, but I don't think we currently support those anyway.

-----

With all that said:
- this change only impacts the attendee registration modal.
- we can modify TC to stop sending formatted floats to the modal, but my understanding is that this would be a larger change that we probably can't fit into gobi. Probably a good change include ET-1331.
- if we're ok with telling non-USD folks to not use the Attendee Registration modal for now, we can remove this piece and merge only the PHP code, and it will work properly with the Attendee Registration page.

Reply:
No idea what it was, but it looks like some change from Gobi had already dealt with this issue, so none of the JS code is needed anymore.